actix-session/src/middleware.rs

```diff
@@ -181,6 +183,25 @@ pub enum SessionLength {
         /// `max_session_length` is also used as the TTL for the session state in the
         /// storage backend.
         max_session_length: Option<Duration>,
+        /// If set to `Some(true)`, the session cookie expiration is reset (i.e. `Max-Age` is set
+        /// to `now + max_session_length`) every time the client associated with the session
+        /// makes a request. The TTL on the session state is updated accordingly.
+        ///
+        /// If set to `Some(false)`, the session cookie expiration is set when the session gets
+        /// created. It is refreshed exclusively when the session data changes or the session key
+        /// is renewed.
+        ///
+        /// If left unspecified (`None`), it defaults to `false`.
+        ///
+        /// # Performance impact
+        ///
+        /// Refreshing the expiration on activity is not free.
+        /// It is not enough to refresh the session cookie expiration, we need to refresh the TTL
+        /// on the session state as well. This translates into a request over the network if you
+        /// are using a remote system as storage backend (e.g. Redis).
+        /// This impacts both the total load on your storage backend (i.e. number of
+        /// queries it has to handle) and the latency of the requests served by your server.
+        refresh_session_ttl_when_active: Option<bool>,
```

Comment:
Why not just `bool`?

Reply:
Actually, bool is not very descriptive, and enum would be much better. Either an enum to describe auto-refresh option or a third option in the `SessionLength` enum.

Reply:
I don't think that a third enum variant in `SessionLength` is the right approach - this only makes sense for persistent session tokens, which is why I nested into this variant.

Using an `Option` is a way to give the user the opportunity to set only some parameters within the variant while saying "I trust you for the rest". I can probably make that more explicit by having a separate type that gets nested inside the variant rather than its fields inlined.

I agree we could use an enum instead of a boolean.